Demo/DemoChat/Sources/UI/TextToSpeechView.swift

```diff
@@ -0,0 +1,155 @@
+                        Task {
+                            await store.createSpeech(query)
+                        }
+                        prompt = ""
+                    }
+                    .foregroundColor(.accentColor)
+                    .disabled(prompt.replacingOccurrences(of: " ", with: "").isEmpty)
+                    Spacer()
+                }
+            }
+            if !$store.audioObjects.wrappedValue.isEmpty {
+                Section("Click to play, swipe to save:") {
+                    ForEach(store.audioObjects) { object in
+                        HStack {
+                            Text(object.prompt.capitalized)
+                            Spacer()
+                            Button(action: {
+                                guard let player = object.audioPlayer,
+                                        object.format != AudioSpeechQuery.AudioSpeechResponseFormat.opus.rawValue else { return }
+                                
+                                if player.isPlaying {
+                                    player.stop()
+                                } else {
+                                    player.prepareToPlay()
+                                    player.volume = 1
+                                    player.play()
+                                }
+                            }, label: {
+                                Image(systemName: "play.fill").foregroundStyle(object.format == AudioSpeechQuery.AudioSpeechResponseFormat.opus.rawValue ? Color.secondary : Color.accentColor)
+                            })
+                        }
+                        .swipeActions(edge: .trailing, allowsFullSwipe: false) {
+                            Button {
+                                presentUserDirectoryDocumentPicker(for: object.originResponse.audioData!, filename: "GeneratedAudio.\(object.format)")
+                            } label: {
+                                Image(systemName: "square.and.arrow.down")
+                            }
+                            .tint(.accentColor)
+                        }
+                    }
+                }
+            }
+        }
+        .listStyle(.insetGrouped)
+        .scrollDismissesKeyboard(.interactively)
+        .navigationTitle("Create Speech")
+    }
+    
+}
+
+extension TextToSpeechView {
+    
+    private func presentUserDirectoryDocumentPicker(for audioData: Data, filename: String) {
+        store.getFileInDocumentsDirectory(audioData, fileName: filename) { fileUrl in
+            let filePickerVC = UIDocumentPickerViewController(forExporting: [fileUrl], asCopy: false)
+            filePickerVC.shouldShowFileExtensions = true
+            
+            guard let vc = getCurrentViewController() else { return }
+            vc.present(filePickerVC, animated: true, completion: nil)
+        }
+    }
+    
+    private func getCurrentViewController() -> UIViewController? {
```

Comment:
What do you think about moving it as extension for `View`? As for me, it is great solution not only for `TextToSpeechView`